include/crab/domains/symbolic_variable_eq_domain.hpp

```diff
@@ -0,0 +1,1031 @@
+    equivalence_class_elems_t left_equiv_classes =
+        (*this).equiv_classes_elems();
+    equivalence_class_elems_t right_equiv_classes = e.equiv_classes_elems();
+    bool res = true;
+    for (auto &right_kv : right_equiv_classes) {
+      const element_set_t &right_set = right_kv.second;
+      CRAB_LOG("symb-var-equiv-classes",
+               print_elems_vector(crab::outs(), right_set););
+      assert(right_set.size() > 0);
+      bool tmp = false;
+      for (auto &left_kv : left_equiv_classes) {
+        const element_set_t &left_set = left_kv.second;
+        CRAB_LOG("symb-var-equiv-classes",
+                 print_elems_vector(crab::outs(), left_set););
+        // subset relations implies the dual of domain operation \sqsubseteq
+        tmp |= std::includes(left_set.begin(), left_set.end(),
+                             right_set.begin(), right_set.end());
+      }
+      res &= tmp;
+    }
+    return res;
+  }
+
+  this_domain_t operator|(const this_domain_t &e) const {
+    SVEQ_DOMAIN_SCOPED_STATS(".join");
+    if (is_bottom()) {
+      return e;
+    } else if (e.is_bottom()) {
+      return *this;
+    } else if (is_top() || e.is_top()) {
+      this_domain_t res;
+      return res;
+    } else {
+      return join(*this, e);
+    }
+  }
+
+  void operator|=(const this_domain_t &e) {
+    SVEQ_DOMAIN_SCOPED_STATS(".join");
+    if (is_bottom()) {
+      *this = e;
+    } else if (e.is_bottom()) {
+      return;
+    } else if (is_top() || e.is_top()) {
+      set_to_top();
+    } else {
+      *this = join(*this, e);
+    }
+  }
+
+  this_domain_t operator&(const this_domain_t &e) const {
+    SVEQ_DOMAIN_SCOPED_STATS(".meet");
+    if (is_bottom() || e.is_top()) {
+      return *this;
+    } else if (e.is_bottom() || is_top()) {
+      return e;
+    } else {
+      return meet(*this, e);
+    }
+  }
+
+  this_domain_t operator||(const this_domain_t &e) const {
+    SVEQ_DOMAIN_SCOPED_STATS(".widening");
+    return *this;
+  }
+
+  this_domain_t operator&&(const this_domain_t &e) const {
+    SVEQ_DOMAIN_SCOPED_STATS(".narrowing");
+    return *this;
+  }
+
+  /// @brief expand what x equals to to y. This is equivalent to add
+  /// @param x the orginal variable in some class
+  /// @param y
+  void expand(const element_t &x, const element_t &y) {
+    if (is_bottom() || is_top() || !contains(x)) {
+      return;
+    }
+    add(x, y);
+  }
+
+  /// @brief assign equality, this is equivalent to add y to x's equiv cls
+  /// @param x an element
+  /// @param y an element
+  void assign(const element_t &x, const element_t &y) {
+    if (!is_bottom()) {
+      add(x, y);
+    }
+  }
+
+  /// @brief assign bool lhs := rhs, not used
+  /// @param lhs an element
+  /// @param rhs an element
+  /// @param is_not_rhs if true the assignment is lhs := not(rhs)
+  void assign_bool_var(const element_t &lhs, const element_t &rhs,
+                       bool is_not_rhs) {
+    // WARN: this should not happen
```

Comment:
Add `SASSERT(false && "this domain does not treat equalities between boolean vars")`

But why shouldn't it happen? Maybe for the object domain it is not expected to happen.